lib/manager/docker-compose/extract.spec.ts

```diff
@@ -31,5 +32,21 @@ describe('manager/docker-compose/extract', () => {
       expect(res.deps).toMatchSnapshot();
       expect(res.deps).toHaveLength(8);
     });
+    it('extracts default variable values for version 3', () => {
+      const res = extractPackageFile(yamlFile3DefaultValue);
+      expect(res.deps).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+            "currentDigest": "sha256:abcd",
+            "currentValue": "5.0.0",
+            "datasource": "docker",
+            "depName": "redis",
+            "replaceString": "\${REDIS_IMAGE:-redis:5.0.0@sha256:abcd}",
```

Comment:
Have you run this on a real repo? I would not have expected this to work with the above `autoReplaceStringTemplate`

Reply:
Hi @rarkins, you were right, it didn't replace the string correctly. After my fixes, it seems to work on my test repo https://github.com/olegkrivtsov/renovate-test-repo/blob/master/docker-compose.yml

```
 INFO: Repository started (repository=olegkrivtsov/renovate-test-repo)
       "renovateVersion": "0.0.0-semantic-release"
 INFO: Dependency extraction complete (repository=olegkrivtsov/renovate-test-repo)
       "baseBranch": "master",
       "stats": {
         "managers": {"docker-compose": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
 INFO: Branch updated (repository=olegkrivtsov/renovate-test-repo, branch=renovate/redis-6.x)
       "commitSha": "d7157ab"
 INFO: PR created (repository=olegkrivtsov/renovate-test-repo, branch=renovate/redis-6.x)
       "pr": 8,
       "prTitle": "Update redis Docker tag to v6"
 INFO: Issue created (repository=olegkrivtsov/renovate-test-repo)
 INFO: Deleting orphan branch (repository=olegkrivtsov/renovate-test-repo, branch=renovate/redis-5.0.0)
 INFO: Deleting orphan branch (repository=olegkrivtsov/renovate-test-repo, branch=renovate/redis-5.x)
 INFO: Repository finished (repository=olegkrivtsov/renovate-test-repo)
       "durationMs": 24123
Done in 52.40s.
```

It created the following PR, which looks correct: https://github.com/olegkrivtsov/renovate-test-repo/blob/master/docker-compose.yml